test/common/upstream/health_checker_impl_test.cc

```diff
@@ -614,6 +614,51 @@ TEST_F(HttpHealthCheckerImplTest, RemoteCloseBetweenChecks) {
   EXPECT_TRUE(cluster_->hosts_[0]->healthy());
 }
 
+TEST_F(HttpHealthCheckerImplTest, StreamReachesWatermarkDuringCheck) {
+  setupNoServiceValidationHC();
+  EXPECT_CALL(*this, onHostStatus(_, false));
+
+  cluster_->hosts_ = {makeTestHost(cluster_->info_, "tcp://127.0.0.1:80")};
+  expectSessionCreate();
+  expectStreamCreate(0);
+  EXPECT_CALL(*test_sessions_[0]->timeout_timer_, enableTimer(_));
+  health_checker_->start();
+
+  EXPECT_CALL(*test_sessions_[0]->interval_timer_, enableTimer(_));
+  EXPECT_CALL(*test_sessions_[0]->timeout_timer_, disableTimer());
+
+  test_sessions_[0]->request_encoder_.stream_.runHighWatermarkCallbacks();
+  test_sessions_[0]->request_encoder_.stream_.runLowWatermarkCallbacks();
+
+  respond(0, "200", true);
+  EXPECT_TRUE(cluster_->hosts_[0]->healthy());
+}
+
+TEST_F(HttpHealthCheckerImplTest, ConnectionReachesWatermarkDuringCheck) {
+  setupNoServiceValidationHC();
+  EXPECT_CALL(*this, onHostStatus(_, false));
+
+  cluster_->hosts_ = {makeTestHost(cluster_->info_, "tcp://127.0.0.1:80")};
+  expectSessionCreate();
+  expectStreamCreate(0);
+  EXPECT_CALL(*test_sessions_[0]->timeout_timer_, enableTimer(_));
+  health_checker_->start();
+
+  EXPECT_CALL(*test_sessions_[0]->interval_timer_, enableTimer(_));
+  EXPECT_CALL(*test_sessions_[0]->timeout_timer_, disableTimer());
+
+  for (auto* callback : test_sessions_[0]->client_connection_->callbacks_) {
```

Comment:
Sorry can you do same thing here also? Thanks. (There are already event helpers IIRC).

Reply:
Oops, forgot about those ones.

Reply:
I found helpers in the `ConnectionImpl` class but not in the mock class

Reply:
https://github.com/envoyproxy/envoy/pull/1699/files#diff-e96d2141e2131be66d0db09f98bd4a16R32. Can you move there as it applies to both types of connections?